typings/glamorous.d.ts

```diff
@@ -36,18 +50,18 @@ export interface Config {
   useDisplayNameInClassName: boolean
 }
 
-export interface GlamorousInterface extends HTMLGlamorousInterface, SVGGlamorousInterface {
+export interface GlamorousInterface extends HTMLComponentFactory, SVGComponentFactory {
   <Props>(
     component:Component<Props>,
-    options?: GlamorousOptions,
-  ): StyledFunction<Props, CSSProperties>
+    options?: Partial<GlamorousOptions>,
+  ): GlamorousComponentFactory<Props, CSSProperties>
   <Props>(
     component:Component<Props>,
-    options?: GlamorousOptions,
-  ): StyledFunction<Props, SVGProperties>
+    options?: Partial<GlamorousOptions>,
+  ): GlamorousComponentFactory<Props, SVGProperties>
```

Comment:
should also add support for all the html/svg tags https://github.com/wooorm/html-tag-names/blob/master/index.json and https://github.com/wooorm/svg-tag-names/blob/master/index.json